imp/TokenSet.cs

```diff
@@ -176,18 +176,36 @@ private TokenSet()
         } /* end TokenSet */
 
         /* ---------------------------------------------------------------------------
-        * private constructor TokenSet (uint[])
-        * ---------------------------------------------------------------------------
-        * used to instantiate the prefabricated first and follow lists
+         * private TokenSet newFromRawData(uint[])
+         * ---------------------------------------------------------------------------
+         * Used to instantiate the prefabricated first and follow lists.
+         * Expects TSB[segmentCount] to be the number of tokens, and returns null if 
+         * the count does not match.
          * ------------------------------------------------------------------------ */
 
-        public TokenSet(params uint[] TSB)
+        public static TokenSet newFromRawData(params uint[] TSB)

```

Comment:
TSB is a non-descriptive name, also it goes against C# naming conventions to have a parameter in all-uppercase. Since the variadic argument list includes both the bit patterns and a bit counter, there isn't really any single name that will be specific, so we have to settle for a generic name. I generally use `args` in such cases.